### DISCUSSION TITLE
CWL Version Supported

### DISCUSSION THREAD

**Discussion by pcampanella** [EOEPCA/eoepca#83] [ADES (Processing)] [answered]
We are working to deploy WASDI Applications in ADES. 
We produce a CWL file that can be validated within CWL 1.2 and 1.1
It does not validate with 1.0 because we need the network access.
We did not found in the documentation a specific requirement on this (maybe our fault).
If we run locally the CWL it works (but we need to authenticate to the registry, even if this is another point).

When we try to deploy on ADES we receive:
{"title":"NoApplicableCode","type":"NoApplicableCode","detail":"Unable to run the Service. The message returned back by the Service was the following: Tool definition failed validation: no workflow or command line tool was found in cwl file."}

Is the CWL Version fixed?  Can we suppose the system can work with version 1.2 ( https://www.commonwl.org/v1.2/CommandLineTool.html  ) ?

**Comment by rconway** [comment]
@bbrauzzi Can you confirm the version of CWL that is supported in the ADES.
Also, can you confirm the validation that takes place within the ADES during deployment - i.e. what tool/library is used to make this validation. It might be useful to know from the perspective of a user - to check their own CWL.

**Comment by rconway** [comment, marked as answer]
Calrissian supports CWL 1.0, 1.1, 1.2 as described in the conformance statement here - https://github.com/Terradue/calrissian/tree/conformance-1.2.1/conformance
